internal/services/aadgraph/application_app_role_resource.go

```diff
@@ -184,34 +184,57 @@ func applicationAppRoleResourceRead(d *schema.ResourceData, meta interface{}) er
 			d.SetId("")
 			return nil
 		}
-		return fmt.Errorf("retrieving Application ID %q: %+v", id.ObjectId, err)
+		return tf.ErrorDiag(fmt.Sprintf("Retrieving Application with ID %q", id.ObjectId), err.Error(), "application_object_id")
+	}
+
+	role, err := graph.AppRoleFindById(app, id.RoleId)
+	if err != nil {
+		return tf.ErrorDiag("Identifying App Role", err.Error(), "")
 	}
 
-	role := graph.AppRoleFindById(app, id.RoleId)
 	if role == nil {
 		log.Printf("[DEBUG] App Role %q (ID %q) was not found - removing from state!", id.RoleId, id.ObjectId)
 		d.SetId("")
 		return nil
 	}
 
-	d.Set("application_object_id", id.ObjectId)
-	d.Set("role_id", id.RoleId)
-	d.Set("allowed_member_types", role.AllowedMemberTypes)
-	d.Set("description", role.Description)
-	d.Set("display_name", role.DisplayName)
-	d.Set("is_enabled", role.IsEnabled)
-	d.Set("value", role.Value)
+	if err := d.Set("application_object_id", id.ObjectId); err != nil {
+		return tf.ErrorDiag("Could not set attribute", err.Error(), "application_object_id")
+	}
```

Comment:
We now need to check for simple properties?? if so maybe we could write a helper:
```suggestion
	if diag := tf.Set("application_object_id", id.ObjectId); err != nil {
		return diag
	}
```

Reply:
I'm not certain whether we _have_ to, but it makes the linter happy and I figured it was good practice? Thanks for the tip, have implemented that and it tidies it up nicely :+1: